### PR TITLE
Update Restore-PSPester to use 'Save-Module' to get the latest version of Pester

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Update Restore-PSPester to use 'Save-Module' to get the latest version of Pester.
We have moved to the latest version of Pester via PR #6064

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Review by adityapatwardhan — APPROVED** [review]
LGTM

File: docs/testing-guidelines/testing-guidelines.md [hunk 1]
```diff
@@ -106,15 +106,15 @@ Currently, we have a minuscule number of tests which are run by using xUnit.
...
+Two helper functions are part of the build.psm1 module to help with that:
 
-* `Restore-PSPester` will restore Pester, which is needed to run `Start-PSPester`
 * `Start-PSPester` will execute all Pester tests which are run by the CI system
 * `Start-PSxUnit` will execute the available xUnit tests run by the CI system
 
 Our CI system runs these as well; there should be no difference between running these on your dev system, versus in CI.
 
-Make sure that you run `Restore-PSPester` before running `Start-PSPester`, or it will fail to run.
+Make sure that you run `Save-Module -Name Pester -Path $PSHOME\Modules` before running `Start-PSPester`
```
Comment (TravisEz13) [inline docs/testing-guidelines/testing-guidelines.md]: This isn't quite accurate.  It should be the $PSHome of the process that is started by Start-PSPester.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

**Review by JamesWTruher — APPROVED** [review]
I think that Travis' comment is reasonable, it's one of the reasons I left it in the earlier functions, as `$PSHOME` is the right `$PSHOME` in the circumstance where it would be run.

**Comment by TravisEz13** [comment]
It might be easier to update the Restore-PSPester to do the correct action.

**Comment by daxian-dbw** [comment]
Update the PR to just change `Restore-PSPester` to use `Save-Module` within it.

**Review by TravisEz13 — APPROVED** [review]

**Review by JamesWTruher — APPROVED** [review]

**Comment by iSazonov** [comment]
@daxian-dbw Could you please correct PR header and description?

**Comment by daxian-dbw** [comment]
@iSazonov Good catch. Updated.

**Pull request merged**